doc/cprover-manual/goto-harness.md

```diff
@@ -0,0 +1,263 @@
+[CPROVER Manual TOC](../../)
+
+## Goto Harness
+
+This is a tool that can generate harnesses, that is functions that instrument
+another function under analysis, by setting up an appropriate environment before
+calling them.
+
+This is useful when trying to analyse an isolated unit of a program without
+having to manually construct an appropriate environment.
+
+### Usage
+
+We have two types of harnesses we can generate for now:
+
+* The `memory-snapshot` harness. TODO: Daniel can document this.
```

Comment:
There's more than one 'Daniel'....

Reply:
Maybe explicitly reference the PR (#4261)?